config/confighttp/confighttp_test.go

```diff
@@ -817,13 +817,14 @@ func TestHttpCors(t *testing.T) {
 				}))
 			require.NoError(t, err)
 
+			waitStart := make(chan struct{})
 			go func() {
+				close(waitStart)
 				_ = s.Serve(ln)
 			}()
 
-			// TODO: make starting server deterministic
 			// Wait for the servers to start
-			<-time.After(10 * time.Millisecond)
+			<-waitStart
 
```

Comment:
We usually do a WaitGroup for go routines, and I am not saying this is not correct. But let's be consistent and use that everywhere, simplifies things.

Also, this is still not guaranteed to have the server started unfortunately.

Reply:
ok sure thing

Reply:
I think there are some good guarantees this will work. I have tried and reproduced the scenario with this Go playground: https://go.dev/play/p/yrBE49_-cbk
I have based my work off this SO thread: https://stackoverflow.com/questions/32738188/how-can-i-start-the-browser-after-the-server-started-listening/32742904#32742904

Reply:
Based on those findings, I can even try to remove all the time.Sleep statements as we shouldn't have to wait at all.